app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryimage/EntryImageActivity.java

```diff
@@ -165,6 +167,24 @@ protected void onSubmitButtonClicked() {
         presenter.onEntrySubmitted();
     }
 
+    @Override
+    public void showInvalidEntryErrorMessage() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(R.string.error_creating_entry)
+                .setTitle(R.string.error_creating_entry_title)
+                .setNeutralButton(R.string.common_ok, (dialog, id) -> {
+                });
+        builder.create().show();
+    }
+
+    @Override
+    public void showApiErrorMessage() {
+        Context context = getApplicationContext();
+        int duration = Toast.LENGTH_SHORT;
+        Toast toast = Toast.makeText(context, getResources().getString(R.string.error_api), duration);
```

Comment:
where this is an activity you can replace the getApplicationContext and context stuff with just "this"